compute/create_instance.go

```
@@ -0,0 +1,99 @@
+func createInstance(w io.Writer, projectID string, zone string, instanceName string, machineType string, sourceImage string, networkName string) error {
+	// projectID := "your_project_id"
+	// zone := "europe-central2-b"
+	// instanceName := "your_instance_name"
+	// machineType := "n1-standard-1"
+	// sourceImage := "projects/debian-cloud/global/images/family/debian-10"
+	// networkName := "global/networks/default"
+
+	ctx := context.Background()
+	instancesClient, err := compute.NewInstancesRESTClient(ctx)
+	if err != nil {
+		return fmt.Errorf("NewInstancesRESTClient: %v", err)
+	}
+	defer instancesClient.Close()
+
+	req := &computepb.InsertInstanceRequest{
+		Project: projectID,
+		Zone:    zone,
+		InstanceResource: &computepb.Instance{
+			Name: proto.String(instanceName),
+			Disks: []*computepb.AttachedDisk{
+				&computepb.AttachedDisk{
+					InitializeParams: &computepb.AttachedDiskInitializeParams{
+						DiskSizeGb:  proto.Int64(10),
+						SourceImage: proto.String(sourceImage),
+					},
+					AutoDelete: proto.Bool(true),
+					Boot:       proto.Bool(true),
+					Type:       computepb.AttachedDisk_PERSISTENT.Enum(),
+				},
+			},
+			MachineType: proto.String(fmt.Sprintf("zones/%s/machineTypes/%s", zone, machineType)),
+			NetworkInterfaces: []*computepb.NetworkInterface{
+				&computepb.NetworkInterface{
+					Name: proto.String(networkName),
+				},
+			},
+		},
+	}
+
+	op, err := instancesClient.Insert(ctx, req)
+	if err != nil {
+		return fmt.Errorf("Create instances request: %v", err)
+	}
+
+	if op.GetStatus() == computepb.Operation_RUNNING {
+		zoneOperationsClient, err := compute.NewZoneOperationsRESTClient(ctx)
+		if err != nil {
+			return fmt.Errorf("NewZoneOperationsRESTClient: %v", err)
+		}
+		defer zoneOperationsClient.Close()
+
+		req := &computepb.WaitZoneOperationRequest{
+			Operation: op.GetName(),
+			Project:   projectID,
+			Zone:      zone,
+		}
+
+		zoneOperationsClient.Wait(ctx, req)
```

This returns an operation and an error. The error is not being checked. This applies to other LRO operations as well.